src/slurm_plugin/fleet_manager.py

```diff
@@ -0,0 +1,151 @@
+        return "{class_name}({attrs})".format(class_name=self.__class__.__name__, attrs=attrs)
+
+    def _get_config(self, config_file_path):
+        """Get fleetmanager configuration."""
+        log.info("Reading %s", config_file_path)
+
+        config = ConfigParser()
+        try:
+            config.read_file(open(config_file_path, "r"))
+        except IOError:
+            log.error(f"Cannot read slurm fleet manager configuration file: {config_file_path}")
+            raise
+
+        self.region = config.get("slurm_fleet_manager", "region")
+        self.cluster_name = config.get("slurm_fleet_manager", "cluster_name")
+        self.terminate_max_batch_size = config.getint(
+            "slurm_fleet_manager", "terminate_max_batch_size", fallback=self.DEFAULTS.get("terminate_max_batch_size")
+        )
+        # Configure boto3 to retry 1 times by default
+        self._boto3_retry = config.getint("slurm_fleet_manager", "boto3_retry", fallback=self.DEFAULTS.get("max_retry"))
+        self._boto3_config = {"retries": {"max_attempts": self._boto3_retry, "mode": "standard"}}
+        proxy = config.get("slurm_fleet_manager", "proxy", fallback=self.DEFAULTS.get("proxy"))
+        if proxy != "NONE":
+            self._boto3_config["proxies"] = {"https": proxy}
+        self.boto3_config = Config(**self._boto3_config)
+
+        self.logging_config = config.get(
+            "slurm_fleet_manager", "logging_config", fallback=self.DEFAULTS.get("logging_config")
+        )
+
+        log.info(self.__repr__())
+
+
+def _fleet_manager(config, computefleet_status_data_path):
+    computefleet_status = _get_computefleet_status(computefleet_status_data_path)
+
+    if ComputeFleetStatus.is_stop_requested(computefleet_status):
+        _stop_partition(config)
+    elif ComputeFleetStatus.is_start_requested(computefleet_status):
+        _start_partition()
+
+
+def _start_partition():
+    log.info("Setting slurm partition to UP and resuming nodes...")
+    update_all_partitions(PartitionStatus.UP, reset_node_addrs_hostname=False)
+    resume_powering_down_nodes()
+
+
+def _stop_partition(config):
+    log.info("Setting slurm partition to INACTIVE and terminating all compute nodes...")
+    update_all_partitions(PartitionStatus.INACTIVE, reset_node_addrs_hostname=True)
+    instance_manager = InstanceManager(
+        config.region,
+        config.cluster_name,
+        config.boto3_config,
+    )
+    instance_manager.terminate_all_compute_nodes(config.terminate_max_batch_size)
+
+
+def _get_computefleet_status(computefleet_status_data_path):
+    try:
+        with open(computefleet_status_data_path, "r", encoding="utf-8") as computefleet_status_data_file:
+            computefleet_status = ComputeFleetStatus(
+                json.load(computefleet_status_data_file).get(ComputeFleetStatusManager.COMPUTE_FLEET_STATUS_ATTRIBUTE)
+            )
+        log.info("ComputeFleet status is: %s", computefleet_status)
+    except IOError:
```

Comment:
Would it make sense to handle the broader Exception (e.g. what if the JSON is malformed)?